module-2/backend/week_one.md

```diff
@@ -1,32 +1,96 @@
+    @count = 1
+    @name = "Mr. Ed"
     erb :index
   end
   ```
 
-8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed` to the view?
-9. What's the purpose of ERB?
-10. Why do I need a development AND test database?
-11. What is CRUD and why is it important?
-12. What does HTTP stand for? 
-13. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
-14. What's an ORM?
-15. What's the most commonly used ORM in ruby (Sinatra & Rails)?
-16. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
-17. What's a migration? 
-18. When you create a migration, does it automatically modify your database?
-19. How does a model relate to a database?
-20. What is the difference between `#new` and `#create`?
+7. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed` to the view?  
+  See above.
+
+8. What's the purpose of ERB?
+  The purpose of the ERB above is to tell the controller which page to display if the user navigates
+  to a certain URI.  The use of ERB in the view is to allow Ruby objects/attributes to be displayed on
+  the view.
+
+9. Why do I need a development AND test database?
+  So that
+10. What is CRUD and why is it important?
+  CREATE READ UPDATE DELETE
+
+11. What does HTTP stand for?
+  Hyper Text Transfer Protocol
+
+12. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+  With <%= erb.object %> or <% erb.object %>.  The first one would display on the page, the second
+  would not.
+
+13. What's an ORM?
+  Object Relational Mapping
+  Used to convert data from incompatible systems into objects so that they can be treated within a programming language as compatible sources
+
+14. What's the most commonly used ORM in ruby (Sinatra & Rails)?
+  ActiveRecordcd
+
+15. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
+
+  get '/restaurants'  --> erb :index
+  To show the index page of all restaurant records
+
+  get '/restaurants/new' --> erb :new
+  post '/restaurants' --> redirect "/restaurants/#{id}"
+  To create a new restaurant record in database
+
+  get '/restaurants/:id' --> erb :show
+  To show a restaurant's record on page
+
+  get '/restaurants/edit' --> erb :edit
+  put '/restaurants/:id' --> redirect "/restaurants/#{id}"
+  To edit a restaurant record
+
+  delete '/restaurants/:id' --> redirect '/restaurants'
+  To delete a restaurant record
+
+16. What's a migration?
+  A migration is transferring data from one source to another.  In our case so far, from a CSV

```

Comment:
Migrations are directions for how to create the structure of your database. In order to actually change the structure of the database you need to run the migration.

Seeding, on the other hand, populates your database.